modules/qr_code.py

```diff
@@ -0,0 +1,55 @@
+import qrcode
+import uuid
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+class QRCode:
+    def __init__(self, base_url, qr_cache, max_size) -> None:
+        self.mapping = {}
+        self.base_url = base_url
+        self.qr_cache = qr_cache
+        self.max_size = max_size
+
+    def add(self, alias: str):
+        try:     
+            url = f"{self.base_url}/{alias}" 
+            filename = f"{uuid.uuid4()}.jpg"
+            path = os.path.join(self.qr_cache, filename)
+
+            img = qrcode.make(url)
+            img.save(path)
+            self.mapping[alias] = path
+
+            if len(self.mapping) > self.max_size: #removes files if exceeds size
+                remove_alias,remove_path = self.mapping.popitem()
+                os.remove(remove_path)
+                logger.debug(f"Removed qrcode with alias: {remove_alias} to free space.")
+
+            return path
+        except FileNotFoundError:
+            logger.exception(f"Could not find folder {self.qr_cache}:")
+        except OSError:
+            logger.exception(f"Error occured when handling files:")
+        except Exception:
+            logger.exception(f"An unexpected error occured")
+
+    def find(self, alias: str):
+        try: 
+            return self.mapping[alias]
+        except KeyError:
+            logger.exception(f"Couldn't find qr code with alias {alias}: ")
+            return None
```

Comment:
we can avoid the `KeyError` all together
```suggestion
        return self.mapping.get(alias)
```